dicom-archive/updateMRI_Upload.pl

```diff
@@ -126,64 +145,55 @@
 ################################################################
 
 # fetch tarchiveLibraryDir from ConfigSettings in the database
-my $tarchiveLibraryDir = &NeuroDB::DBI::getConfigSetting(
-                            \$dbh,'tarchiveLibraryDir'
-                            );
+
+my $configOB = NeuroDB::objectBroker::ConfigOB->new(db => $db);
+my $tarchiveLibraryDir = $configOB->getTarchiveLibraryDir();
+
 # determine tarchive path stored in the database (without tarchiveLibraryDir)
 my $tarchive_path = $tarchive;
-$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;  
+$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;
 
-my $where         = " WHERE t.ArchiveLocation =?";
-
-if ($globArchiveLocation) {
-    $where = " WHERE t.ArchiveLocation LIKE ?";
-    $tarchive_path = basename($tarchive);
-}
+# Check if there is already an mri upload record for the tarchive
+my $mriUploadOB = NeuroDB::objectBroker::MriUploadOB->new(db => $db);
+my $resultRef = $mriUploadOB->getWithTarchive(
+    GET_COUNT, $tarchive_path, $globArchiveLocation
+);
 
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  COUNT(*) 
-FROM 
-  mri_upload m 
-JOIN 
-  tarchive t ON (t.TarchiveID=m.TarchiveID) 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute($tarchive_path);
-my $count = $sth->fetchrow_array;
-if($count>0) {
-   print "\n\tERROR: the tarchive is already uploaded \n\n"; 
+if($resultRef->[0]->[0] > 0) {
+   print "\n\tERROR: the tarchive is already uploaded \n\n";
    exit 6;
-} 
-
+}
 
 ################################################################
 #####get the tarchiveid from tarchive table#####################
 ################################################################
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  t.TarchiveID 
-FROM 
-  tarchive t 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute("%".$tarchive_path."%");
-my $tarchiveID = $sth->fetchrow_array;
 
+my $tarchiveOB = NeuroDB::objectBroker::TarchiveOB->new(db => $db);
+$resultRef = $tarchiveOB->getByTarchiveLocation(
+    ['TarchiveID'], $tarchive_path, $globArchiveLocation
+);
+
+if(@$resultRef != 1) {
+    die sprintf(
+        "Unexpected number of tarchive records with location %s found: %d\n",
+        $tarchive_path,
+        scalar(@$resultRef)
+    );
+}
+my $tarchiveID = $resultRef->[0]->[0];
 
 ################################################################
  #####populate the mri_upload columns with the correct values####
 ################################################################
-($query = <<QUERY) =~ s/\n/ /gm;
-INSERT INTO mri_upload 
-  (UploadedBy, UploadDate, TarchiveID, DecompressedLocation) 
-VALUES
-  (?, now(), ?, ?)
-QUERY
-my $mri_upload_insert = $dbh->prepare($query);
-$mri_upload_insert->execute($User,$tarchiveID,$source_location);
+
+$mriUploadOB->insert(
```

Comment:
just out of curiosity: @cmadjar @nicolasbrossard 
I do not use this script because I do not run dicomTar.pl directly, but for projects that do, how do you populate the isPhantom field?
as I was testing @nicolasbrossard 's pull request, I added the option `-updateMRI_upload` to running dicomTar.pl from imaging_upload_file.pl, and that gave weird results! 
nothing to do with this pull request, but just wondering how projects insert into the mri_upload table Phantom entries using dicomTar.pl directly? I am suspecting all those entered are defaulting to isPhantom='N'.

Reply:
Indeed, I just checked and they are all  isPhantom='N' in PREVENT-AD... :S

Reply:
o'o'
anyway I created a redmine ticket (RM #14102) about it, besides another bug as reported earlier (dual mri_upload entries)...